pkg/dao/snapshots.go

```diff
@@ -78,3 +84,19 @@ func (sDao snapshotDaoImpl) Delete(snapUUID string) error {
 	}
 	return nil
 }
+
+func (sDao snapshotDaoImpl) FetchLatestSnapshot(repoConfigUUID string) (api.SnapshotResponse, error) {
+	var snap models.Snapshot
+	result := sDao.db.
+		Joins("inner join repository_configurations on repository_configurations.repository_uuid = snapshots.repository_uuid").
+		Where("repository_configurations.uuid = ?", repoConfigUUID).
+		Where("snapshots.org_id = repository_configurations.org_id").
+		Order("created_at DESC").
+		First(&snap)
```

Comment:
Andrew's pr makes this a lot simpler by adding a latest_snapshot_uuid column to repo_config, but this is good for now